pkg/plugin/config.go

```diff
@@ -15,13 +15,66 @@
 // Package plugin provides the implementation of the JVS plugin interface.
 package plugin
 
+import (
+	"errors"
+	"fmt"
+
+	"github.com/abcxyz/pkg/cli"
+)
+
 // PluginConfig defines the set over environment variables required
 // for running the plugin.
 type PluginConfig struct {
 	// ID of the GitHub APP we use to authenticate.
-	GitHubAppID string `env:"GITHUB_APP_ID,required"`
+	GitHubAppID string
 	// Installation ID of the github app.
-	GitHubAppInstallationID string `env:"GITHUB_APP_INSTALLATION_ID,required"`
+	GitHubAppInstallationID string
 	// The private Key PEM obtained for github app.
-	GitHubAppPrivateKeyPEM string `env:"GITHUB_APP_PRIVATE_KEY_PEM,required"`
+	GitHubAppPrivateKeyPEM string
+}
+
+// Validate validates if the config is valid.
+func (cfg *PluginConfig) Validate() error {
+	var rErr error
+	if cfg.GitHubAppID == "" {
+		rErr = errors.Join(rErr, fmt.Errorf("GITHUB_APP_ID is empty"))
+	}
+	if cfg.GitHubAppInstallationID == "" {
+		rErr = errors.Join(rErr, fmt.Errorf("GITHUB_APP_INSTALLATION_ID is empty"))
+	}
+	if cfg.GitHubAppPrivateKeyPEM == "" {
```

Comment:
The PEM is only checked if it's empty here. As for checking if the pem can be decoded, I think it should be the job of `cli/server.go` which will be implemented in the next PR, since `server.go` will use the private key to create `githubapp.Client` and use that for creating new validator. (https://github.com/abcxyz/jvs-plugin-github/blob/main/pkg/plugin/plugin.go#L53)

Reply:
It's helpful to think about the abstractions and interfaces across the boundary. If we expect `cli/server.go` to get a private key as an input, it would be best to use an [AfterParse function](https://github.com/abcxyz/pkg/blob/032652eef237fb544150b642e9d6f2171b3a2eec/cli/command_doc_after_parse_test.go#L67) here to parse the key material and store the result on the config as a private field on the Go structure.

If, however, we expect cli/server.go to accept a PEM-encoded string, then it makes sense to push this downstream.

Reply:
Backgroud info: the private key users got from github app is a `private-key.pem` file, and we will just upload that to secret manager and mount to cloud run.

Therefore,  for one, I think the server expects the PEM-encoded string. 

For two, if we handle the decoding in `config.go`.
 
1. Parsing pem into `*rsa.PrivateKey` is a kind of validation, and it should be done in `Validation()` and not by `AfterParse` in `ToFlags()`.
2.  However, if we validate the pem can be decoded in `Validation()` and set the privatekey, it's also a bit weird as setting fields are not part of `Validation()`'s job.
3.  People can just do `c := &PluginConfig{}`, and run the validation, therefore, even we parse the key in `ToFlags()`, we still need to validate pem here again. So it seems redundant.

Therefore, I think it's best to push it do `cli/server.go` to decode pem, and if it fails, we just don't start the service.

Reply:
3\. Seems like the most compelling reason to me. I agree that it's a bit strange for `Validate` to modify the structure, but it's the best option here.